test/e2e/openHelpAbout.test.ts

```diff
@@ -1,38 +1,46 @@
-/// <reference types="jest-playwright-preset" />
+import { test, expect } from "@playwright/test"
 import { CODE_SERVER_ADDRESS, STORAGE } from "../utils/constants"
 
-describe("Open Help > About", () => {
-  beforeEach(async () => {
-    // Create a new context with the saved storage state
-    // so we don't have to logged in
+test.describe("Open Help > About", () => {
+  // Create a new context with the saved storage state
+  // so we don't have to logged in
+  const options: any = {}
+  // TODO@jsjoeio
+  // Fix this once https://github.com/microsoft/playwright-test/issues/240
+  // is fixed
+  if (STORAGE) {
     const storageState = JSON.parse(STORAGE) || {}
-    await jestPlaywright.resetContext({
+    options.contextOptions = {
       storageState,
-    })
-    await page.goto(CODE_SERVER_ADDRESS, { waitUntil: "networkidle" })
-  })
+    }
+  }
 
-  it("should see a 'Help' then 'About' button in the Application Menu that opens a dialog", async () => {
-    // Make sure the editor actually loaded
-    expect(await page.isVisible("div.monaco-workbench"))
+  test(
+    "should see a 'Help' then 'About' button in the Application Menu that opens a dialog",
+    options,
+    async ({ page }) => {
+      await page.goto(CODE_SERVER_ADDRESS, { waitUntil: "networkidle" })
+      // Make sure the editor actually loaded
+      expect(await page.isVisible("div.monaco-workbench"))
 
-    // Click the Application menu
-    await page.click("[aria-label='Application Menu']")
-    // See the Help button
-    const helpButton = "a.action-menu-item span[aria-label='Help']"
-    expect(await page.isVisible(helpButton))
+      // Click the Application menu
+      await page.click("[aria-label='Application Menu']")
```

Comment:
do people still do [page objects](https://martinfowler.com/bliki/PageObject.html) or is that not cool anymore? also, is it possible to select by IDs instead of other attributes, or is this the recommended approach with Playwright?

Reply:
Lol not sure about page objects. Unfortunately the way the HTML is structured, there isn't an ID to use as a selector. Otherwise, we would:
![image](https://user-images.githubusercontent.com/3806031/114921820-d7e04880-9ddf-11eb-8c7c-cfb928c269e2.png)

```html
<div class="menubar-menu-button" role="menuitem" tabindex="0" aria-label="Application Menu" title="Application Menu" aria-haspopup="true" style="visibility: visible;"><div class="menubar-menu-title toolbar-toggle-more codicon codicon-menubar-more" role="none" aria-hidden="true"></div></div>
```

And this HTML comes from VS Code. We could add our own IDs, but there's a chance they get overwritten during a git subtree update to `lib/vscode`. 

This means that if the VS Code Team changes the HTML (i.e. removes the aria-label) then our tests break. This can be annoying for us, but it's the best approach. Plus, aria-labels are less likely to change as frequently as something like a class so at least this attribute is a bit more stable.